src/core/lib/gprpp/posix/directory.cc

```diff
@@ -0,0 +1,99 @@
+//
+//
+// Copyright 2023 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//
+
+#include <grpc/support/port_platform.h>
+
+#include "absl/status/status.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/string_view.h"
+
+#if defined(GPR_LINUX) || defined(GPR_ANDROID) || defined(GPR_FREEBSD) || \
+    defined(GPR_APPLE)
+
+#include <dirent.h>
+#include <sys/stat.h>
+
+#include <string>
+#include <vector>
+
+#include <grpc/support/log.h>
+
+#include "src/core/lib/gprpp/directory.h"
+
+namespace grpc_core {
+
+namespace {
+std::string GetAbsoluteFilePath(absl::string_view valid_file_dir,
+                                absl::string_view file_entry_name) {
+  return absl::StrFormat("%s/%s", valid_file_dir, file_entry_name);
+}
+}  // namespace
+
+absl::StatusOr<std::vector<std::string>> Directory::GetFilesInDirectory() {
+  DIR* directory;
+  // Open the dir for reading
+  if ((directory = opendir(directory_path_.c_str())) == nullptr) {
+    return absl::InternalError("Could not read crl directory.");
+  }
+  std::vector<std::string> files;
+  struct dirent* directory_entry;
+  // Iterate over everything in the directory
+  while ((directory_entry = readdir(directory)) != nullptr) {
+    const char* file_name = directory_entry->d_name;
+
+    std::string file_path =
+        GetAbsoluteFilePath(directory_path_.c_str(), file_name);
+    struct stat dir_entry_stat;
+    int stat_return = stat(file_path.c_str(), &dir_entry_stat);
```

Comment:
Ok that makes sense

I changed the method to `GetDirectoryContents` and now explicitly just `continue` on `.` and `..`

I'll need a little bit of time to map this to Windows, because right now I'm using `FindFirstFile` and associated functions, so I just need to go read up on the docs for this to see what to use.